src/sentences/alm.rs

```diff
@@ -0,0 +1,224 @@
+use core::{fmt::Debug, ops::RangeInclusive, str};
+use nom::{
+    character::{
+        complete::{char, hex_digit0},
+        streaming::hex_digit1,
+    },
+    combinator::{map_res, opt},
+    IResult,
+};
+
+use crate::{Error, NmeaSentence, SentenceType};
+
+use super::utils::number;
+
+/// ALM - GPS Almanac Data
+///
+/// <https://gpsd.gitlab.io/gpsd/NMEA.html#_alm_gps_almanac_data>
+/// ```text
+///         1   2   3  4   5  6    7  8    9    10     11     12     13     14  15   16
+///         |   |   |  |   |  |    |  |    |    |      |      |      |      |   |    |
+///  $--ALM,x.x,x.x,xx,x.x,hh,hhhh,hh,hhhh,hhhh,hhhhhh,hhhhhh,hhhhhh,hhhhhh,hhh,hhh*hh<CR><LF>
+/// ```
+///  
+///  Field Number:
+///  
+///  1. Total number of messages
+///  2. Sentence Number
+///  3. Satellite PRN number (01 to 32)
+///  4. GPS Week Number (range 0 to 2^13 - 1), where:
+///     - 0 is the week of the GPS Week Number epoch on January 6th 1980;
+///     - 8191 is the week that precedes the next rollover on January 6th 2137;
+///  6. Eccentricity
+///  7. Almanac Reference Time
+///  8. Inclination Angle
+///  9. Rate of Right Ascension
+/// 10. Root of semi-major axis
+/// 11. Argument of perigee
+/// 12. Longitude of ascension node
+/// 13. Mean anomaly
+/// 14. F0 Clock Parameter
+/// 15. F1 Clock Parameter
+/// 16. Checksum
+///  
+///  Fields 5 through 15 are dumped as raw hex.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct AlmData {
+    total_number_of_messages: Option<u16>,
+    sentence_number: Option<u16>,
+    satellite_prn_number: Option<u8>,
+    gps_week_number: Option<u16>,
+    sv_health: Option<u8>,
+    eccentricity: Option<u16>,
+    almanac_reference_time: Option<u8>,
+    inclination_angle: Option<u16>,
+    rate_of_right_ascension: Option<u16>,
+    root_of_semi_major_axis: Option<u32>,
+    argument_of_perigee: Option<u32>,
+    longitude_of_ascension_node: Option<u32>,
+    mean_anomaly: Option<u32>,
+    f0_clock_parameter: Option<u16>,
+    f1_clock_parameter: Option<u16>,
+}
+
+pub fn parse_alm(sentence: NmeaSentence) -> Result<AlmData, Error> {
+    if sentence.message_id != SentenceType::ALM {
+        Err(Error::WrongSentenceHeader {
+            expected: SentenceType::ALM,
+            found: sentence.message_id,
+        })
+    } else {
+        Ok(do_parse_alm(sentence.data)?.1)
+    }
+}
+
+fn number_in_range<T>(i: &str, range: RangeInclusive<T>) -> IResult<&str, T>
+where
+    T: str::FromStr + PartialOrd + Debug,
+{
+    map_res(number::<T>, |number_str| {
+        if range.contains(&number_str) {
+            return Ok(number_str);
+        }
+        Err(format!(
```

Comment:
`format!` is only available when we have `std` or `alloc`.
We have used static &str in other places so I'm not sure how we can format this one.
I'm open to suggestions

Reply:
I think `nom` will return the error string, so I can just change the text to be `"Parsed number is outside of the expected range"`?